uitest/Pages/DashboardPage.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using DevHome.UITest.Dialogs;
+using DevHome.UITest.Extensions;
+using OpenQA.Selenium.Appium;
+using OpenQA.Selenium.Appium.Windows;
+
+namespace DevHome.UITest.Pages;
+
+/// <summary>
+/// Page model class for the dashboard page
+/// </summary>
+public class DashboardPage : ApplicationPage
+{
+    private WindowsElement AddWidgetButton => Driver.FindElementByAccessibilityId("AddWidgetButton");
+
+    private IReadOnlyCollection<WindowsElement> WidgetItems => Driver.FindElementsByAccessibilityId("WidgetItem");
+
+    public IReadOnlyCollection<WidgetControl> DisplayedWidgets => WidgetItems.Select(i => new WidgetControl(Driver, i)).ToList();
+
+    public DashboardPage(WindowsDriver<WindowsElement> driver)
+        : base(driver)
+    {
+    }
+
+    public AddWidgetDialog ClickAddWidgetButton()
+    {
+        AddWidgetButton.Click();
+        return new AddWidgetDialog(Driver, this);
+    }
+
+    /// <summary>
+    /// Remove all dashboard widgets one by one
+    /// </summary>
+    public void RemoveAllWidgets()
+    {
+        // Remove widgets in reverse order. It is slower to remove widgets
+        // top-down because the remove animation could cause the automated
+        // cursor to miss the 'more options' button, often requiring two
+        // attempts to remove a single widget
+        foreach (var widget in DisplayedWidgets.Reverse())
+        {
+            widget.Remove();
+        }
+    }
+
+    public void WaitForWidgetsToBeLoaded()
+    {
+        // Wait for the progress ring to disappear
+        Driver.WaitUntilInvisible(ByWindowsAutomation.AccessibilityId("LoadingWidgetsProgressRing"));
+    }
+
+    /// <summary>
+    /// Control model class for a widget on the dashboard
+    /// </summary>
+    public class WidgetControl
```

Comment:
Would it make sense to have this in its own file?

Reply:
I prefer to keep this an internal class as it only represents a widget control displayed on the Dashboard, and may not necessary be compatible for representing widget controls on other pages in the future.